beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -203,23 +198,14 @@ impl<E: EthSpec> PendingComponents<E> {
     ///
     /// Returns `true` if both the block exists and the number of received blobs / custody columns
     /// matches the number of expected blobs / custody columns.
-    pub fn is_available(&self, block_import_requirement: &BlockImportRequirement) -> bool {
-        match block_import_requirement {
-            BlockImportRequirement::AllBlobs => self
-                .num_expected_blobs()
-                .map_or(false, |num_expected_blobs| {
-                    num_expected_blobs == self.num_received_blobs()
-                }),
-            BlockImportRequirement::CustodyColumns(num_expected_columns) => {
-                let num_received_data_columns = self.num_received_data_columns();
-                // No data columns when there are 0 blobs
-                self.num_expected_blobs()
-                    .map_or(false, |num_expected_blobs| {
-                        num_expected_blobs == 0
-                            || *num_expected_columns == num_received_data_columns
-                    })
-            }
-        }
+    pub fn is_available(&self, custody_column_count: usize) -> bool {
+        self.num_expected_blobs()
+            .map_or(false, |num_expected_blobs| {
+                let all_blobs_received = num_expected_blobs == self.num_received_blobs();
+                let all_columns_received = num_expected_blobs == 0
+                    || custody_column_count == self.num_received_data_columns();
```

Comment:
beacon_chain code seems to imply that we can have too many columns, should this be
```rust
self.num_received_data_columns() >= custody_column_count
```

Reply:
How is this possible? only custody & sample columns should get in here, if we have other random columns getting into da checker, this check won't work.

Reply:
https://github.com/sigp/lighthouse/pull/6268/files#r1764455920

Reply:
ah ok what see what you mean, we can change that check to `==`

Reply:
Addressed in 0a9d5a0f61b4114dff93fe88c46ca4b97dde75f2